src/Common/createFile.cpp

```diff
@@ -0,0 +1,50 @@
+#include "createFile.h"
+#include <Common/ErrorCodes.h>
+#include <Common/Exception.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int FILE_ALREADY_EXISTS;
+extern const int OPERATION_NOT_PERMITTED;
+extern const int NOT_ENOUGH_SPACE;
+extern const int CANNOT_CREATE_FILE;
+}
+}
+
+namespace FS
+{
+[[noreturn]] void handleLastError(const std::string & path)
+{
+    switch (errno)
+    {
+        case EEXIST:
+            throw DB::Exception(DB::ErrorCodes::FILE_ALREADY_EXISTS, "File {} already exist", path);
+        case EPERM:
+            throw DB::Exception(DB::ErrorCodes::OPERATION_NOT_PERMITTED, "Not enough permissions to create file {}", path);
+        case ENOSPC:
+            throw DB::Exception(DB::ErrorCodes::NOT_ENOUGH_SPACE, "Not enough space to create file {}", path);
+        case ENAMETOOLONG:
+            throw DB::Exception(DB::ErrorCodes::CANNOT_CREATE_FILE, "File name {} is too long");
+        default:
+            throw DB::Exception(DB::ErrorCodes::CANNOT_CREATE_FILE, "Cannot create file {}. Error: {}", path, strerror(errno));
+    }
+}
+
+/// Copy from Poco::createFile
+bool createFile(const std::string & path)
+{
+    int n = open(path.c_str(), O_WRONLY | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH);
+    if (n != -1)
+    {
+        close(n);
+        return true;
+    }
+    handleLastError(path);
```

Comment:
Maybe just `DB::throwFromErrnoWithPath(...)`?

Reply:
But it requires DB::ErrorCode to be already known when calling `DB::throwFromErrnoWithPath(message, path, db_error_code, the_errno = errno)` , and in this place DB::ErrorCode is taken according to errno, so it will anyway require switch with errno cases to deternine DB error code.

Reply:
We can call it like this: `throwFromErrnoWithPath("Cannot create file " + path, path, ErrorCodes::CANNOT_CREATE_FILE)`. It's ok to use general `CANNOT_CREATE_FILE` for all filesystem errors here.

It will add return value of `errnoToString(code, the_errno)` (calls `strerror` internally) to exception message. It will handle all possible values of errno.